### PR TITLE
Worker improvements

### PR DESCRIPTION
**Pull request by ronnix**
Multiple changes conspiring together to avoid the situation where an initial fetch is blocked behind a queue of periodic updates...

### CODE REVIEW

File: repondeur/zam_repondeur/tasks/fetch.py [hunk 1]
```diff
@@ -14,10 +14,12 @@
...
 
-@huey.task(retries=3, retry_delay=60 * 5)  # Minutes.
-@huey.lock_task("fetch-articles-lock")
+RETRY_DELAY = 5 * 60  # 5 minutes
+
+
+@huey.task(retries=3, retry_delay=RETRY_DELAY)
 def fetch_articles(lecture_pk: int) -> None:
-    with transaction.manager:
+    with transaction.manager, huey.lock_task(f"fetch-articles-{lecture_pk}"):
```
Comment (davidbgk) [inline repondeur/zam_repondeur/tasks/fetch.py]: 👌

File: repondeur/zam_repondeur/tasks/periodic.py [hunk 2]
```diff
@@ -23,4 +23,5 @@ def update_data() -> None:
 def fetch_all_amendements() -> None:
     with transaction.manager:
         for lecture in DBSession.query(Lecture):
-            fetch_amendements(lecture.pk)
+            delay = lecture.pk % (15 * 60)  # spread out updates over 15 minutes
```
Comment (davidbgk) [inline repondeur/zam_repondeur/tasks/periodic.py]: I'm not sure we should rely on the lecture's pk, maybe safer to enumerate on lectures and compute the delay based on an truly incremental index?

Comment (ronnix) [reply repondeur/zam_repondeur/tasks/periodic.py]: The idea was that using the `lecture_pk` it allows each lecture to be updated every hour, with a stable offset. What do you think ?

**Comment by ronnix** [comment]
It's more work to make the periodic task configurable, because it's defined at import time in the decorator... ☹️

**Comment by davidbgk** [comment]
[Indeed](https://huey.readthedocs.io/en/latest/api.html#Huey.periodic_task), maybe coding our custom [crontab](https://huey.readthedocs.io/en/latest/api.html#crontab) method which relies on settings but I'm not even sure we can delay that 🤔

File: repondeur/zam_repondeur/tasks/periodic.py (hunk 2 above)
Comment (ronnix) [reply repondeur/zam_repondeur/tasks/periodic.py]: I guess I'm tired 🙂

**Pull request merged**